deepchem/molnet/load_function/zinc15_datasets.py

```diff
@@ -0,0 +1,252 @@
+  format : str (default '2D')
+    Format of data to download. 2D SMILES strings or 3D SDF files.
+  featurizer : allowed featurizers for this dataset
+    A featurizer that inherits from deepchem.feat.Featurizer.
+  transformers : List of allowed transformers for this dataset
+    A transformer that inherits from deepchem.trans.Transformer.
+  splitter : allowed splitters for this dataset
+    A splitter that inherits from deepchem.splits.splitters.Splitter.
+  reload : bool (default True)
+    Try to reload dataset from disk if already downloaded. Save to disk
+    after featurizing.
+  data_dir : str, optional (default None)
+    Path to datasets.
+  save_dir : str, optional (default None)
+    Path to featurized datasets.
+  featurizer_kwargs : dict
+    Specify parameters to featurizer, e.g. {"size": 1024}
+  splitter_kwargs : dict
+    Specify parameters to splitter, e.g. {"seed": 42}
+  transformer_kwargs : dict
+    Maps transformer names to constructor arguments, e.g.
+    {"BalancingTransformer": {"transform_x":True, "transform_y":False}}
+  dataset_size : str (default '250K')
+    Number of compounds to download; '250K', '1M', '10M', or '270M'.
+  dataset_dimension : str (default '2D')
+    SMILES strings (2D) or 3D SDF files; '2D' or '3D'
+  test_run : bool (default False)
+    Flag to indicate tests, if True dataset is not downloaded.
+  **kwargs : additional optional arguments.
+
+  Returns
+  -------
+  tasks, datasets, transformers : tuple
+    tasks : list
+      Column names corresponding to machine learning target variables.
+    datasets : tuple
+      train, validation, test splits of data as
+      ``deepchem.data.datasets.Dataset`` instances.
+    transformers : list
+      ``deepchem.trans.transformers.Transformer`` instances applied
+      to dataset.
+
+  Notes
+  -----
+  The total ZINC dataset with SMILES strings contains hundreds of millions
+  of compounds and is over 100GB! ZINC250K is recommended for experimentation.
+  The full set of 270M goldilocks compounds is 23GB.
+
+  References
+  ----------
+  .. [1] Sterling and Irwin. J. Chem. Inf. Model, 2015 http://pubs.acs.org/doi/abs/10.1021/acs.jcim.5b00559.
+
+  Examples
+  --------
+  >>> import deepchem as dc
+  >>> tasks, datasets, transformers = dc.molnet.load_zinc15(test_run=True)
+  >>> train_dataset, val_dataset, test_dataset = datasets
+  >>> n_tasks = len(tasks)
+  >>> n_features = train_dataset.X.shape[1]
+  >>> model = dc.models.MultitaskRegressor(n_tasks, n_features)
+
+  """
+
+  # Featurize zinc15
+  logger.info("About to featurize zinc15.")
+  my_tasks = ['mwt', 'logp', 'reactive']  # machine learning targets
+
+  if test_run:
+    ds = deepchem.data.NumpyDataset(np.zeros((10, 1)))
+    return my_tasks, (ds, ds, ds), []
+
+  # Raise warnings and list other available options
+  if dataset_size not in ['250K', '1M', '10M', '270M']:
+    raise ValueError("""
+      Only '250K', '1M', and '10M' are supported for dataset_size.
```

Comment:
Typo: "Only '250K', '1M', '10M',  and '270M' are..."

Reply:
Fixed, thanks!